systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,272 @@
+                    return false;
+                }
+            })
+            .collect(Collectors.toList());
+    }
+
+    public static void waitForTopicWithPrefixDeletion(String namespaceName, String topicPrefix, int start, int end) {
+        TestUtils.waitFor("deletion of all topics with prefix: " + topicPrefix + " from " + start + " to " + end,
+            TestConstants.GLOBAL_POLL_INTERVAL, TestConstants.GLOBAL_TIMEOUT,
+            () -> {
+                try {
+                    final int numberOfTopicsToDelete = getAllKafkaTopicsWithPrefix(namespaceName, topicPrefix, start, end).size();
+                    LOGGER.info("Remaining KafkaTopic's to delete from {} to {}: {} !", start, end, numberOfTopicsToDelete);
+                    return numberOfTopicsToDelete == 0;
+                } catch (Exception e) {
+                    LOGGER.error("Error while checking for topic deletion: ", e);
+                    return e.getMessage().contains("Not Found") || e.getMessage().contains("the server doesn't have a resource type");
+                }
+            });
+    }
+
+    /**
+     * Manages the full lifecycle of Kafka topics concurrently using a fixed thread pool.
+     * This method processes creation, modification, and deletion for each topic in separate threads,
+     * ensuring all topics are handled simultaneously across available CPU resources.
+     *
+     * +-----------------------------------------------------------------------------+
+     * |                            processAllTopicsConcurrently                     |
+     * +-----------------------------------------------------------------------------+
+     * | +-------------+       +-------------+             +-------------+           |
+     * | | Topic 1     |       | Topic 2     |     ...     | Topic N     |           |
+     * | | +---------+ |       | +---------+ |             | +---------+ |           |
+     * | | | Thread  | |       | | Thread  | |             | | Thread  | |           |
+     * | | |         | |       | |         | |             | |         | |           |
+     * | | | Creation| |       | | Creation| |             | | Creation| |           |
+     * | | | Update  | |       | | Update  | |             | | Update  | |           |
+     * | | | Deletion| |       | | Deletion| |             | | Deletion| |           |
+     * | | +---------+ |       | +---------+ |             | +---------+ |           |
+     * | +-------------+       +-------------+             +-------------+           |
+     * +-----------------------------------------------------------------------------+
+     *
+     * @param testStorage       An instance of TestStorage containing configuration and state needed for topic operations.
+     * @param numberOfTopics    The number of Kafka topics to be processed.
+     */
+    public static void processAllTopicsConcurrently(TestStorage testStorage, int numberOfTopics) {
```

Comment:
Should this be part of the test? It's part of the logic and it's only used there.

Reply:
That's premature optimization :)